swift/BUILD.bazel

```diff
@@ -0,0 +1,65 @@
+load("@rules_pkg//:mappings.bzl", "pkg_attributes", "pkg_filegroup", "pkg_files")
+load("@rules_pkg//:install.bzl", "pkg_install")
+load("@ql//:defs.bzl", "codeql_platform")
+load("@ql_utils//:source_dir.bzl", "current_source_dir")
+
+pkg_files(
+    name = "dbscheme",
+    srcs = [
+        "ql/lib/swift.dbscheme",
+        "ql/lib/swift.dbscheme.stats",
+    ],
+)
+
+pkg_files(
+    name = "qltest",
+    srcs = ["tools/qltest.sh"],
+    attributes = pkg_attributes(mode = "0755"),
+    prefix = "tools",
+)
+
+pkg_files(
+    name = "manifest",
+    srcs = ["codeql-extractor.yml"],
+)
+
+pkg_filegroup(
+    name = "extractor-pack-generic",
+    srcs = [
+        ":dbscheme",
+        ":manifest",
+        ":qltest",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+pkg_files(
+    name = "extractor",
+    srcs = ["//swift/extractor"],
+    attributes = pkg_attributes(mode = "0755"),
+    prefix = "tools/" + codeql_platform,
+)
+
+pkg_filegroup(
+    name = "extractor-pack-arch",
+    srcs = [":extractor"],
+    visibility = ["//visibility:public"],
+)
+
+pkg_filegroup(
+    name = "extractor-pack",
+    srcs = [
+        ":extractor-pack-arch",
+        ":extractor-pack-generic",
+    ],
+    visibility = ["//visibility:public"],
+)
+
+pkg_install(
```

Comment:
I wasn't aware of `pkg_install` - it is not documented, and it's replacement, `pkg_tree`, is not ready yet: https://github.com/bazelbuild/rules_pkg/issues/388.